lib/topics_filter.rb

```diff
@@ -1,12 +1,63 @@
 # frozen_string_literal: true
 
 class TopicsFilter
-  def initialize(guardian:, scope: Topic, category_id: nil)
+  def initialize(guardian:, scope: Topic)
     @guardian = guardian
     @scope = scope
-    @category = category_id.present? ? Category.find_by(id: category_id) : nil
   end
 
+  def filter_from_query_string(query_string)
+    return @scope if query_string.blank?
+
+    query_string.scan(/(?<exclude>-)?(?<key>\w+):(?<value>[^:\s]+)/) do |exclude, key, value|
+      case key
+      when "status"
+        @scope = filter_status(status: value)
+      when "tags"
+        value.scan(
+          /^(?<tags>([a-zA-Z0-9\-]+)(?<delimiter>[,+])?([a-zA-Z0-9\-]+)?(\k<delimiter>[a-zA-Z0-9\-]+)*)$/,
+        ) do |value, delimiter|
+          match_all =
+            if delimiter == ","
```

Comment:
minor but maybe `match_all = delimeter == "+"`

Reply:
Good point :+1: Let me update it.

Reply:
O sorry this doesn't work because when there are no delimiters we want `match_all` to default to `true` as well.